### PR TITLE
FEATURE: QA-18068 Parallelization for ui-sdk cypress test

### PR DESCRIPTION
**Pull request by maitrieuphilong**
JIRA: QA-18068

---

Supported PR commands:

| Command                                                 | Description                                                |
| ------------------------------------------------------- | ---------------------------------------------------------- |
| `ok to test`                                            | Re-run standard checks                                     |
| `extended check sonar`                                  | SonarQube tests                                            |
| `extended test - backstop`                              | BackstopJS tests                                           |
| **E2E Cypress tests commands - TIGER**                  |                                                            |
| `extended test - tiger-cypress - isolated <testName>`   | Run isolated tests running against recorded Tiger backend. |
| `extended test - tiger-cypress - record <testName>`     | Create a new recording for isolated Tiger tests.           |
| `extended test - tiger-cypress - integrated <testName>` | Run integrated tests against live backend                  |
| **E2E Cypress tests commands - BEAR**                   |                                                            |
| `extended test - cypress - isolated <testName>`         | Run isolated tests running against recorded Bear backend.  |
| `extended test - cypress - record <testName>`           | Create a new recording for isolated Bear tests.            |
| `extended test - cypress - integrated <testName>`       | Run integrated tests against live backend                  |

`<testName>` in cypress commands is used to filter specfiles. Example, to run record with BEAR backend

-   Against `dashboard.spec.ts` and `drilling.spec.ts`, execute command `extended test - cypress - record dashboard,drilling`
-   Against all specfiles, execute command `extended test - cypress - record` or `extended test - cypress - record *`

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `extended test - backstop` passes
-   [ ] `extended test - tiger-cypress - record` to record new mapping files (Tiger BE)
-   [ ] `extended test - cypress - record` to record new mapping files (Bear BE)
-   [ ] `extended test - tiger-cypress - isolated` passes
-   [ ] `extended test - cypress - isolated` passes
-   [ ] `extended test - tiger-cypress - integrated` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/12116/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/14082/ : FAILURE in 5m 57s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/12117/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/14083/ : SUCCESS in 15m 31s

File: libs/sdk-ui-tests-e2e/README.md [hunk 1]
```diff
@@ -146,3 +146,18 @@ and run with (by default runs in visual mode)
...
+By default, cypress is running in serial mode. To enable parallelization, [Currents](https://currents.dev/readme/getting-started/you-first-cypress-run) is integrated into cypress.
+To use Currents, `RUN_PARALLEL` must be set to `true` and following environemnts must be set
+
+```
+export CURRENTS_PROJECT_ID
+export CYPRESS_RECORD_KEY
+export CURRENTS_CI_BUILD_ID
+```
+
+-   `CURRENTS_PROJECT_ID` is unique for each UI-app respository, for development purpose is `4wrwOa`
```
Comment (caohhung) [inline libs/sdk-ui-tests-e2e/README.md:161]: This is not correct dev PROJECT_ID, please check https://docs.google.com/document/d/1tRJhhwiuFX5PvASGRSgqp1HJfMSK2aCS7CQwGi_aYXM/edit#heading=h.rhybfzm6mz3x

**Review by kandl — APPROVED** [review]
LGTM

**Comment by maitrieuphilong** [comment]
extended test - cypress - isolated

File: libs/sdk-ui-tests-e2e/README.md (hunk 1 above)
Comment (maitrieuphilong) [reply libs/sdk-ui-tests-e2e/README.md:161]: updated

**Comment by gdgate** [comment]
Build in pipeline check-extended aborted.

**Comment by maitrieuphilong** [comment]
extended test - cypress - isolated

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/12132/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/14092/ : SUCCESS in 15m 50s

**Comment by yenkins** [comment]
# ✨ Cypress test results ✨

## 🎉 Great news! Tests were successful.

|Status|Name|Tests|Failures|Time|
|:--:|:--|:--:|:--:|--:|
|✅|**dashboardDependentFilter.spec.ts**|8|-|85.4250s|
|✅|**singleSelectionAttributeFilterParentChild.spec.ts**|2|-|11.1770s|
|✅|**pivotTableAggregationsMenuAndAddTotal.spec.ts**|5|-|17.8630s|
|✅|**attributeFilterButtonParentChild.spec.ts**|3|-|12.9360s|
|✅|**dateFilter.spec.ts**|9|-|11.8400s|
|✅|**dashboard.spec.ts**|9|-|56.2300s|
|✅|**pivotTableRemoveAllFromMutipleAggregation.spec.ts**|1|-|6.0800s|
|✅|**dashboardDnD.spec.ts**|7|-|54.9970s|
|✅|**parentChildFiltersWithPlaceholders.spec.ts**|3|-|15.7980s|
|✅|**barChartDrilling.spec.ts**|4|-|5.6530s|
|✅|**drilling.spec.ts**|7(2 skipped)|-|36.9170s|
|✅|**dashboardKPI.spec.ts**|8|-|135.8820s|
|✅|**pivotTableRemoveOneFromMutipleAggregation.spec.ts**|1|-|6.3110s|
|✅|**singleSelectionAttributeFilterButton.spec.ts**|3|-|7.7320s|
|✅|**pivotTableSizing.spec.ts**|1|-|4.1240s|
|✅|**pivotTableSizingReset.spec.ts**|15|-|35.7390s|
|✅|**attributeFilterButtonWithSelection.spec.ts**|2|-|6.0890s|
|✅|**attributeFilterButton.spec.ts**|13|-|47.7930s|
|✅|**pivotTableRemoveOneAggregation.spec.ts**|1|-|6.1010s|
|✅|**attributeFilter.spec.ts**|1|-|6.0010s|
|✅|**headlineChartDrilling.spec.ts**|2|-|3.0960s|
|✅|**dashboardFiltering.spec.ts**|18(2 skipped)|-|137.1590s|
|✅|**dashboardDateRelated.spec.ts**|8|-|62.9090s|

You can find full report [here](https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-cypress-bear-isolated-zuul/479/).

**Comment by yenkins** [comment]
# ✨ Cypress test results ✨

## 🎉 Great news! Tests were successful.

|Status|Name|Tests|Failures|Time|
|:--:|:--|:--:|:--:|--:|
|✅|**dashboardDependentFilter.spec.ts**|8|-|85.4250s|
|✅|**singleSelectionAttributeFilterParentChild.spec.ts**|2|-|11.1770s|
|✅|**pivotTableAggregationsMenuAndAddTotal.spec.ts**|5|-|17.8630s|
|✅|**attributeFilterButtonParentChild.spec.ts**|3|-|12.9360s|
|✅|**dateFilter.spec.ts**|9|-|11.8400s|
|✅|**dashboard.spec.ts**|9|-|56.2300s|
|✅|**pivotTableRemoveAllFromMutipleAggregation.spec.ts**|1|-|6.0800s|
|✅|**dashboardDnD.spec.ts**|7|-|54.9970s|
|✅|**parentChildFiltersWithPlaceholders.spec.ts**|3|-|15.7980s|
|✅|**barChartDrilling.spec.ts**|4|-|5.6530s|
|✅|**drilling.spec.ts**|7(2 skipped)|-|36.9170s|
|✅|**dashboardKPI.spec.ts**|8|-|135.8820s|
|✅|**pivotTableRemoveOneFromMutipleAggregation.spec.ts**|1|-|6.3110s|
|✅|**singleSelectionAttributeFilterButton.spec.ts**|3|-|7.7320s|
|✅|**pivotTableSizing.spec.ts**|1|-|4.1240s|
|✅|**pivotTableSizingReset.spec.ts**|15|-|35.7390s|
|✅|**attributeFilterButtonWithSelection.spec.ts**|2|-|6.0890s|
|✅|**attributeFilterButton.spec.ts**|13|-|47.7930s|
|✅|**pivotTableRemoveOneAggregation.spec.ts**|1|-|6.1010s|
|✅|**attributeFilter.spec.ts**|1|-|6.0010s|
|✅|**headlineChartDrilling.spec.ts**|2|-|3.0960s|
|✅|**dashboardFiltering.spec.ts**|18(2 skipped)|-|137.1590s|
|✅|**dashboardDateRelated.spec.ts**|8|-|62.9090s|

You can find full report [here](https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-cypress-bear-isolated-zuul/479/).

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/12133/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-pull-request-dispatcher-pipeline/200/ : SUCCESS in 21m 28s

**Review by hungnguyenauto — APPROVED** [review]

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR3583

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Pull request merged**